### ISSUE TITLE
Thin DateTime type in icu_timezone

### ISSUE THREAD

**Issue by Manishearth** [unicode-org/icu4x#5859]
https://github.com/unicode-org/icu4x/issues/5533#issuecomment-2422924363

> Conclusion: Option 4 (Thin DateTime and ZonedDateTime type in icu_timezone), potentially (separate discussion) reexport DateTime from icu_datetime
> 
> Thin `DateTime` type in `icu_timezone`
>  - `icu_calendar` becomes purely `Date`
>  - `icu_timezone` has time zones and times; possibly rename to `icu_time`
>  - IXDTF parsing can return a `DateTime` or `ZonedDateTime`, parser API lives in `icu_timezone`

**Comment by sffc** [comment]
@robertbastian pointed out that in FFI we need to do this

```rust
                .format_any_calendar(&icu_timezone::DateTime {
                    // Arc clone
                    date: date.0.clone(),
                    time: time.0,
                })
```

Possible solutions:

1. Make a private type in FFI that borrows things and implements GetField etc.
2. Option 1 but put it in icu_timezone
3. Change DateTime to be parameterized with AsRef fields

**Comment by Manishearth** [comment]
@sffc I think we have another option: adding `Date::wrap_calendar_in_ref()` that wraps the calendar in a `Ref`, returning a new borrowing date. The rest of Date's fields are a cheap clone and I wouldn't be opposed to guaranteeing that the DateInner types are all Copy. (or at least adding `DateInner: Copy` to `wrap_calendar_in_ref()`

**Comment by sffc** [comment]
Yes that is the correct solution. We solved the ownership problem in the Date type via AsCalendar.

**Comment by robertbastian** [comment]
> icu_timezone has time zones and times; possibly rename to `icu_time`

Further discussion in #5999
